src/StructureFactors/DataRetrieval.jl

```diff
@@ -12,6 +12,179 @@ end
 classical_to_quantum(ω, kT::Float64) = iszero(ω) ? 1.0 : ω/(kT*(1 - exp(-ω/kT)))
 classical_to_quantum(ω, ::Nothing) = 1.0
 
+# Describes a 4D parallelepided histogram in a format compatible with Mantid/Horace
+# 
+# The coordinates of the histogram axes are specified by multiplication 
+# of `q' with each row of the covectors matrix
+mutable struct BinningParameters
+    binstart::SVector{4,Float64}
+    binend::SVector{4,Float64}
+    binwidth::SVector{4,Float64}
+    covectors::SMatrix{4,4,Float64}
+end
+
+function Base.show(io::IO, ::MIME"text/plain", params::BinningParameters)
+    printstyled(io, "Binning Parameters\n"; bold=true, color=:underline)
+    nbin = params.numbins
+    for k = 1:4
+        if nbin[k] == 1
+            printstyled(io, "∫ Integrated"; bold=true)
+        else
+            printstyled(io, @sprintf("⊡ %5d bins",nbin[k]); bold=true)
+        end
+        @printf(io," from %+.3f to %+.3f along [", params.binstart[k], params.binend[k])
+        axes_names = ["x","y","z","E"]
+        inMiddle = false
+        for j = 1:4
+            if params.covectors[k,j] != 0.
+                if(inMiddle)
+                    print(io," ")
+                end
+                @printf(io,"%+.2f d%s",params.covectors[k,j],axes_names[j])
+                inMiddle = true
+            end
+        end
+        @printf(io,"] (Δ = %.3f)", params.binwidth[k]/norm(params.covectors[k,:]))
+        println()
+    end
+end
+
+# Support numbins as a (virtual) property, even though only the binwidth is stored
+Base.getproperty(params::BinningParameters, sym::Symbol) = sym == :numbins ? round.(Int64,(params.binend .- params.binstart) ./ params.binwidth) : getfield(params,sym)
+
+function Base.setproperty!(params::BinningParameters, sym::Symbol, numbins)
+    if sym == :numbins
+        binwidth = (params.binend .- params.binstart) ./ numbins
+        setfield!(params,:binwidth,binwidth)
+    else
+        setfield!(params,sym,numbins)
+    end
+end
+
+# Default coordinates are (Qx,Qy,Qz,ω)
+function BinningParameters(binstart,binend,binwidth;covectors = [1 0 0 0; 0 1 0 0; 0 0 1 0; 0 0 0 1])
+    return BinningParameters(binstart,binend,binwidth,covectors)
+end
+
+function BinningParameters(binstart,binend;numbins,kwargs...)
+    binwidth = (binend .- binstart) ./ numbins
+    return BinningParameters(binstart,binend,binwidth;kwargs...)
+end
+
+# Integrate over one or more axes of the histogram by setting the number of bins
+# in that axis to 1
+function integrate_params!(params::BinningParameters;axis)
```

Comment:
Perhaps `integrate_parameters!`, since "parameters" is written out in full elsewhere. Or maybe `integrate_axes!` is more precise. That it is operating on binning parameters is implied by the argument type.